elastic/integrations
# [TI CIF3] Add IOC Expiration

## Proposed commit message
- Adds IOC expiration as per discussion at https://github.com/csirtgadgets/cif-v5/issues/43#issuecomment-1674988424



## Checklist



## How to test this PR locally
`elastic-package stack down && elastic-package build && elastic-package stack up -d -v && eval "$(elastic-package stack shellinit)" && elastic-package test system --generate  -v`
```
--- Test results for package: ti_cif3 - START ---
╭─────────┬─────────────┬───────────┬──────────────────────────────────────────────────────────────────┬────────┬──────────────╮
│ PACKAGE │ DATA STREAM │ TEST TYPE │ TEST NAME                                                        │ RESULT │ TIME ELAPSED │
├─────────┼─────────────┼───────────┼──────────────────────────────────────────────────────────────────┼────────┼──────────────┤
│ ti_cif3 │             │ asset     │ dashboard ti_cif3-6005a190-0aba-11ed-bcc0-01c79f2670f3 is loaded │ PASS   │        500ns │
│ ti_cif3 │             │ asset     │ dashboard ti_cif3-63a0e470-0a30-11ed-bcc0-01c79f2670f3 is loaded │ PASS   │         83ns │
│ ti_cif3 │             │ asset     │ dashboard ti_cif3-aedada10-0ab5-11ed-bcc0-01c79f2670f3 is loaded │ PASS   │         42ns │
│ ti_cif3 │             │ asset     │ dashboard ti_cif3-b4d9d9b0-0a2f-11ed-bcc0-01c79f2670f3 is loaded │ PASS   │         83ns │
│ ti_cif3 │             │ asset     │ dashboard ti_cif3-bda23600-0abb-11ed-bcc0-01c79f2670f3 is loaded │ PASS   │         42ns │
│ ti_cif3 │             │ asset     │ dashboard ti_cif3-fef149c0-0a2f-11ed-bcc0-01c79f2670f3 is loaded │ PASS   │         83ns │
│ ti_cif3 │ feed        │ asset     │ index_template logs-ti_cif3.feed is loaded                       │ PASS   │        125ns │
│ ti_cif3 │ feed        │ asset     │ ingest_pipeline logs-ti_cif3.feed-1.12.0 is loaded               │ PASS   │         84ns │
╰─────────┴─────────────┴───────────┴──────────────────────────────────────────────────────────────────┴────────┴──────────────╯
--- Test results for package: ti_cif3 - END   ---
Done
Run pipeline tests for the package
--- Test results for package: ti_cif3 - START ---
╭─────────┬─────────────┬───────────┬──────────────────────────────────┬────────┬──────────────╮
│ PACKAGE │ DATA STREAM │ TEST TYPE │ TEST NAME                        │ RESULT │ TIME ELAPSED │
├─────────┼─────────────┼───────────┼──────────────────────────────────┼────────┼──────────────┤
│ ti_cif3 │ feed        │ pipeline  │ test-cif3-no-preserve-ndjson.log │ PASS   │  21.984334ms │
│ ti_cif3 │ feed        │ pipeline  │ test-cif3-sample-ndjson.log      │ PASS   │   3.065291ms │
╰─────────┴─────────────┴───────────┴──────────────────────────────────┴────────┴──────────────╯
--- Test results for package: ti_cif3 - END   ---
Done
Run static tests for the package
--- Test results for package: ti_cif3 - START ---
╭─────────┬─────────────┬───────────┬──────────────────────────┬────────┬──────────────╮
│ PACKAGE │ DATA STREAM │ TEST TYPE │ TEST NAME                │ RESULT │ TIME ELAPSED │
├─────────┼─────────────┼───────────┼──────────────────────────┼────────┼──────────────┤
│ ti_cif3 │ feed        │ static    │ Verify sample_event.json │ PASS   │  53.611292ms │
╰─────────┴─────────────┴───────────┴──────────────────────────┴────────┴──────────────╯
--- Test results for package: ti_cif3 - END   ---
Done
Run system tests for the package
--- Test results for package: ti_cif3 - START ---
╭─────────┬─────────────┬───────────┬───────────┬────────┬───────────────╮
│ PACKAGE │ DATA STREAM │ TEST TYPE │ TEST NAME │ RESULT │  TIME ELAPSED │
├─────────┼─────────────┼───────────┼───────────┼────────┼───────────────┤
│ ti_cif3 │ feed        │ system    │ default   │ PASS   │ 39.578945416s │
╰─────────┴─────────────┴───────────┴───────────┴────────┴───────────────╯
--- Test results for package: ti_cif3 - END   ---
Done
```



## Related issues


- Closes https://github.com/elastic/integrations/issues/7358

## Screenshots


Updated with filter `NOT labels.is_ioc_transform_source: true`
<img width="1724" alt="Screenshot 2024-04-09 at 12 09 42 AM" src="https://github.com/elastic/integrations/assets/11301409/9f6f8002-9174-4cf6-bb77-ba2d875a40ef">
<img width="1724" alt="Screenshot 2024-04-09 at 12 09 21 AM" src="https://github.com/elastic/integrations/assets/11301409/bf38a400-4bc4-4430-9483-9eb350dd56f1">
<img width="1724" alt="Screenshot 2024-04-09 at 12 09 10 AM" src="https://github.com/elastic/integrations/assets/11301409/81234e5d-9f7e-49e0-b465-7219360013ec">
